zipper/types/response.go

```diff
@@ -35,7 +35,7 @@ func (first *ServerInfoResponse) Merge(second *ServerInfoResponse) *errors.Error
 	}
 
 	if first.Err == nil {
-		first.Err = &errors.Errors{}
+		first.Err = new(errors.Errors)
```

Comment:
I'd say use of "new" is generally discouraged and this should be either consistent (e.x. https://dave.cheney.net/2014/08/17/go-has-both-make-and-new-functions-what-gives ) or documented and have very special meaning.

Reply:
If `type A struct{ //... }` then `new(A)` has the same meaning as `&A{}`. It returns a pointer to initialized memory. I agree that if the struct has fields we also want to initialize at the same time, then writing
```
a := &A{
    foo: 10,
    bar: 20,
}
```
is better than:
```
a := new(A)
a.foo = 10
a.bar = 20
```
Here, the zero values for the fields in `Errors` are the only ones we'd want.

> I'd say use of "new" is generally discouraged

I haven't seen that. If you know of discussions where that conclusion is reached I'd be interested in seeing them. Cheney is explaining the difference between two language keywords. `new` gives us a non-nil pointer to a type and it seems fine to use it for that.

Reply:
I'll try to find you a link, but even in the article I gave you Cheney said that seeing "new" usually means that something fishy is going on (in the conclusion).

Of course he is not The Absolute Source Of Truth, but I think that we should have an agreement about that.